### PR TITLE
removed permissioning methods referencing whitelist

### PR DESCRIPTION
**Pull request by macfarla**
## PR description
These methods have been deprecated for years. Long since renamed to allowlist

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PermJsonRpcMethodsTest.java [hunk 1]
```diff
@@ -88,27 +77,4 @@ public void allowlistMethodsPresent() {
     assertThat(rpcMethods.get(PERM_REMOVE_NODES_FROM_ALLOWLIST.getMethodName()))
         .isInstanceOf(PermRemoveNodesFromAllowlist.class);
   }
-
-  @Deprecated
-  @Test
-  public void whitelistMethodsPresent() {
-    final Map<String, JsonRpcMethod> rpcMethods = permJsonRpcMethods.create();
-    assertThat(rpcMethods.size()).isEqualTo(13);
```
Comment (siladu) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PermJsonRpcMethodsTest.java:96]: This assert appears to be missing from allowlistMethodsPresent - if that matters?

**Review by siladu — APPROVED** [review]
LGTM

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/permissioning/PermAddAccountsToWhitelistTest.java [hunk 2]
```diff
@@ -1,137 +0,0 @@
...
-@Deprecated
-@ExtendWith(MockitoExtension.class)
-public class PermAddAccountsToWhitelistTest {
-
-  @Mock private AccountLocalConfigPermissioningController accountWhitelist;
-  private PermAddAccountsToWhitelist method;
-
-  @BeforeEach
-  public void before() {
-    method = new PermAddAccountsToWhitelist(java.util.Optional.of(accountWhitelist));
```
Comment (Matilda-Clerke) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/permissioning/PermAddAccountsToWhitelistTest.java:52]: Should we also remove the classes being tested in these removed tests?

Comment (macfarla) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/permissioning/PermAddAccountsToWhitelistTest.java:52]: YES good spot

Comment (macfarla) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/permissioning/PermAddAccountsToWhitelistTest.java:52]: updated

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PermJsonRpcMethodsTest.java (hunk 1 above)
Comment (macfarla) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PermJsonRpcMethodsTest.java:96]: I don't think that's particularly useful so left it out

**Pull request merged**